server/repository/src/db_diesel/storage_connection.rs

```diff
@@ -19,16 +17,56 @@ const BEGIN_TRANSACTION_STATEMENT: &str = "BEGIN IMMEDIATE;";
 #[cfg(feature = "postgres")]
 const BEGIN_TRANSACTION_STATEMENT: &str = "BEGIN";
 
-pub struct StorageConnection {
-    pub connection: DBConnection,
-    // pub arc_connection: Arc<DBConnection>,
+/// Helper class to avoid deref_mut() calls, which would require to import DerefMut everywhere we
+/// want to use a connection.
+/// For example, with out it it would look like:
+/// let con: &mut DBConnection = LockedConnection.inner.lock().unwrap().deref_mut();
+pub struct LockedConnection<'a> {
+    raw_connection: MutexGuard<'a, DBConnection>,
+}
```

Comment:
Main reason for this LockedConnection was to avoid this deref_mut() problem as described in the comment, i.e. didn't want to manually import DerefMut everywhere.